weave/trace/vals.py

```diff
@@ -202,15 +208,31 @@ def __getattribute__(self, __name: str) -> Any:
         return result
 
     def __setattr__(self, __name: str, __value: Any) -> None:
-        if __name in ["_val", "ref", "server", "root", "mutations"]:
+        if __name in [
+            "_val",
+            "ref",
+            "server",
+            "root",
+            "mutations",
+            "_is_dirty",
+            "parent",
+        ]:
             return object.__setattr__(self, __name, __value)
         else:
+            full_path: tuple[str, ...]
             if not isinstance(self.ref, ObjectRef):
-                raise ValueError("Can only set attributes on object refs")
-            object.__getattribute__(self, "root").add_mutation(
-                self.ref.extra, "setattr", __name, __value
-            )
-            return object.__setattr__(self._val, __name, __value)
+                # We used to raise ValueError here to only set attributes on
```

Comment:
maybe this was because until now we would have expected every single WeaveObject to have a ref?. not clear to me either